backend-rust/src/graphql_api.rs

```diff
@@ -2810,14 +2810,99 @@ impl Account {
 
     async fn transactions(
         &self,
-        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: i32,
+        ctx: &Context<'_>,
+        #[graphql(desc = "Returns the first _n_ elements from the list.")] first: Option<u64>,
         #[graphql(desc = "Returns the elements in the list that come after the specified cursor.")]
-        after: String,
-        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: i32,
+        after: Option<String>,
+        #[graphql(desc = "Returns the last _n_ elements from the list.")] last: Option<u64>,
         #[graphql(desc = "Returns the elements in the list that come before the specified cursor.")]
-        before: String,
+        before: Option<String>,
     ) -> ApiResult<connection::Connection<String, AccountTransactionRelation>> {
-        todo_api!()
+        let config = get_config(ctx)?;
+        let pool = get_pool(ctx)?;
+        let query = ConnectionQuery::<i64>::new(
+            first,
+            after,
+            last,
+            before,
+            config.contract_connection_limit,
+        )?;
+
+        let mut txs = sqlx::query_as!(
+            Transaction,
+            r#"SELECT * FROM (
+                SELECT
+                    index,
+                    block_height,
+                    block_index,
+                    hash,
+                    ccd_cost,
+                    energy_cost,
+                    sender,
+                    type as "tx_type: DbTransactionType",
+                    type_account as "type_account: AccountTransactionType",
+                    type_credential_deployment as "type_credential_deployment: CredentialDeploymentTransactionType",
+                    type_update as "type_update: UpdateTransactionType",
+                    success,
+                    events as "events: sqlx::types::Json<Vec<Event>>",
+                    reject as "reject: sqlx::types::Json<TransactionRejectReason>"
+                FROM transactions
+                WHERE
+                    sender = $1
+                    AND $2 < index
+                    AND index < $3
+                ORDER BY
+                    (CASE WHEN $4 THEN index END) DESC,
+                    (CASE WHEN NOT $4 THEN index END) ASC
+                LIMIT $5
+            ) ORDER BY index ASC
+            "#,
+            self.index,
+            query.from,
+            query.to,
+            query.desc,
+            query.limit as i64,
+        )
+        .fetch(pool);
+
+        let has_previous_page = sqlx::query_scalar!(
+            "SELECT true
+            FROM transactions
+            WHERE sender = $1 AND index <= $2
+            LIMIT 1",
+            self.index,
+            query.from,
+        )
```

Comment:
This isn't super elegant and requires an extra query but I'm not really sure how else to achieve it.

Reply:
If we were to improve, it could be trying to combine some queries to reduce chatter, but this solution should still scale, so I would say this is fine :+1: